storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -128,56 +308,121 @@ private void updateSupervisorsResources(Cluster cluster, Topologies topologies)
-        ret.put(Config.TOPOLOGY_COMPONENT_RESOURCES_OFFHEAP_MEMORY_MB,
-                (Double) _conf.get(Config.TOPOLOGY_COMPONENT_RESOURCES_OFFHEAP_MEMORY_MB));
-        return ret;
+    public User getUser(String user) {
+        return this.userMap.get(user);
+    }
+
+    public Map<String, User> getUserMap() {
+        return this.userMap;
     }
 
     /**
-     * print scheduling for debug purposes
-     * @param cluster
+     * Intialize scheduling and running queues
+     *
      * @param topologies
+     * @param cluster
      */
-    public String printScheduling(Cluster cluster, Topologies topologies) {
-        StringBuilder str = new StringBuilder();
-        Map<String, Map<String, Map<WorkerSlot, Collection<ExecutorDetails>>>> schedulingMap = new HashMap<String, Map<String, Map<WorkerSlot, Collection<ExecutorDetails>>>>();
-        for (TopologyDetails topo : topologies.getTopologies()) {
-            if (cluster.getAssignmentById(topo.getId()) != null) {
-                for (Map.Entry<ExecutorDetails, WorkerSlot> entry : cluster.getAssignmentById(topo.getId()).getExecutorToSlot().entrySet()) {
-                    WorkerSlot slot = entry.getValue();
-                    String nodeId = slot.getNodeId();
-                    ExecutorDetails exec = entry.getKey();
-                    if (schedulingMap.containsKey(nodeId) == false) {
-                        schedulingMap.put(nodeId, new HashMap<String, Map<WorkerSlot, Collection<ExecutorDetails>>>());
-                    }
-                    if (schedulingMap.get(nodeId).containsKey(topo.getId()) == false) {
-                        schedulingMap.get(nodeId).put(topo.getId(), new HashMap<WorkerSlot, Collection<ExecutorDetails>>());
-                    }
-                    if (schedulingMap.get(nodeId).get(topo.getId()).containsKey(slot) == false) {
-                        schedulingMap.get(nodeId).get(topo.getId()).put(slot, new LinkedList<ExecutorDetails>());
-                    }
-                    schedulingMap.get(nodeId).get(topo.getId()).get(slot).add(exec);
+    private void initUsers(Topologies topologies, Cluster cluster) {
+        this.userMap = new HashMap<String, User>();
+        Map<String, Map<String, Double>> userResourcePools = this.getUserResourcePools();
+        LOG.debug("userResourcePools: {}", userResourcePools);
+
+        for (TopologyDetails td : topologies.getTopologies()) {
+            //Get user that submitted topology.  If topology submitter is null or empty string, the topologySubmitter
+            //will be set to anonymous
+            String topologySubmitter = td.getTopologySubmitter();
+            //additional safety check to make sure that topologySubmitter is going to be a valid value
+            if (topologySubmitter == null || topologySubmitter.equals("")) {
+                LOG.error("Cannot determine user for topology {}.  Will skip scheduling this topology", td.getName());
+                continue;
+            }
```

Comment:
If the TopologySubmitter is null or empy string getTopologySubmitter() will return System.getProperty("user.name");". This is consistent with what is done in nimbus.  I guess this is just an additional backup check in case getTopologySubmitter() returned null empty string for some reason.  I will remove the comment.